workflow/sync/throttler.go

```diff
@@ -4,97 +4,80 @@ import (
 	"container/heap"
 	"sync"
 	"time"
-
-	"k8s.io/client-go/util/workqueue"
 )
 
 // Throttler allows CRD controller to limit number of items it is processing in parallel.
 type Throttler interface {
-	Add(key interface{}, priority int32, creationTime time.Time)
+	Add(key string, priority int32, creationTime time.Time)
 	// Next returns true if item should be processed by controller now or return false.
-	Next(key interface{}) (interface{}, bool)
+	Progress(key string) bool
 	// Remove notifies throttler that item processing is done. In responses the throttler triggers processing of previously throttled items.
-	Remove(key interface{})
-	// SetParallelism update throttler parallelism limit.
-	SetParallelism(parallelism int)
```

Comment:
I've removed this which does mandate you restart the controller when you change this config item. This is to reduce complexity-bugs.